routes/api/v1.js

```diff
@@ -424,6 +427,120 @@ api.declare({
   });
 });
 
+api.declare({
+  method:     'get',
+  route:      '/aws/s3/:level/:bucket/:prefix(*)',
+  name:       'awsS3Credentials',
+  input:      undefined,
+  output:     SCHEMA_PREFIX_CONST + 'aws-s3-credentials-response.json#',
+  deferAuth:  true,
+  scopes:     ['auth:aws-s3:<level>:<bucket>/<prefix>'],
+  title:      "Get Temporary Read/Write Credentials S3",
+  description: [
+    "Get temporary AWS credentials for `read-write` or `read-only` access to",
+    "a given `bucket` and `prefix` within that bucket.",
+    "The `level` parameter can be `read-write` or `read-only` and determines",
+    "which type of credentials is returned. Please note that the `level`",
+    "parameter is required in the scope guarding access.",
+    "",
+    "The credentials are set of expire after an hour, but this behavior may be",
+    "subject to change. Hence, you should always read the `expires` property",
+    "from the response, if you intent to maintain active credentials in your",
+    "application.",
+    "",
+    "Please notice that your `prefix` shouldn't start with slash `/`, it is",
+    "allowed for completeness, but it will result in two slashes both in the",
+    "scope and in the bucket. Therefore this pattern is strongly discouraged.",
+    "Also note that if your `prefix` doesn't end in a slash `/` you will not",
+    "be required to insert one. This is mainly a concern when assigning scopes",
+    "to users and doing this right will prevent poor behavior."
+  ].join('\n')
+}, function(req, res) {
+  var level   = req.params.level;
+  var bucket  = req.params.bucket;
+  var prefix  = req.params.prefix;
+
+  // Validate that a proper value was given for level
+  if (level !== 'read-write' && level !== 'read-only') {
```

Comment:
@jhford,
So honestly this is the part that I'm most in doubt about.

Is it really smart to include `read-only` and `read-write` as parameters in the URL. Is that a nice design?